v1/pipeline/stages/cd/deployment-stage-config.yaml

```diff
@@ -1,35 +1,46 @@
+$schema: http://json-schema.org/draft-07/schema#
 title: DeploymentStageConfig
 type: object
 required:
-- execution
+- steps
 properties:
-  customDeploymentRef:
+  custom_deployment_ref:
     $ref: step-template-ref.yaml
-  deploymentType:
+  deployment:
```

Comment:
do we need this in yaml

Reply:
If we remove this how are we going to match service or infra with the deployment then? We are using the deployment type to match with configured service and infra.

Reply:
Isn't the deployment type present inside of the service already, as far as I know, this is just needed for UI do you use this data in the backend anytime?

Reply:
Yes it is used in the backend.

Reply:
We are doing check in back-end code to match the deploymentType with service and infra. Should we remove that logic? If that is the case then i can remove this field.

Reply:
@markoharness I will resolve this conversation but we should really think hard about what we need in YAML and what not, that is the whole point of this exercise. I am not well versed in the area but you should talk about this with SMES here and only include what is absolutely required. @naidu-harness @yogesh-chauhan I would highly recommend you review this PR and include people who you think are required

Reply:
@hextechpal ok. Can we merge this PR as it is and if we decide to remove the backend logic that we have I will open another PR for removing the field?

Reply:
Let's not delay it lets take a call right now it is very hard to revisit stuff. Let's just get the concerned people in a call and close this

Reply:
deploymentType field has been removed.